### PR TITLE
Adding Devcontainer and link to codespace

### PR DESCRIPTION
**Pull request by vncoelho**
With this simpler .devcontainer the project can be opened on codespace and everybody can develop and test things online.
Furthermore, it can help a quick setup on local environment as well.

To test this PR you can click on the link to codespace and pick the "devcontainer" branch, everything will be loaded.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/dotnet
+{
+	"name": "C# (.NET)",
+	"image": "mcr.microsoft.com/devcontainers/dotnet:1-7.0-jammy",
+	"postCreateCommand": "dotnet restore && dotnet build && dotnet test",
```
Comment (shargon) [inline .devcontainer/devcontainer.json:6]: We don't need to test I think:

```suggestion
	"postCreateCommand": "dotnet restore && dotnet build",
```

**Comment by shargon** [comment]
Screenshot_20240110_102637_Chrome

@vncoelho we will need to install some dependencies

**Comment by shargon** [comment]
> > Screenshot_20240110_102637_Chrome
> > 
> > @vncoelho we will need to install some dependencies
> 
> How can I reproduce this, @shargon ?

I just tried to run neo-node

**Comment by vncoelho** [comment]
> > > Screenshot_20240110_102637_Chrome
> > > @vncoelho we will need to install some dependencies
> > 
> > 
> > How can I reproduce this, @shargon ?
> 
> I just tried to run neo-node

This will be different in this case,
It was not yet prepared to run the node...aheuahuea

The first idea was just to introduce the devcontainer for developers to check code, verify tests and etc...
Run the node will probably require other image as well.

**Review by shargon — APPROVED** [review]

**Comment by vncoelho** [comment]
Perfect,@shargon ,
So, lets merge this one and I will open another one for that purpose,which will probably require more discussions.

**Comment by vncoelho** [comment]
I will make some tests with neo-node now

**Comment by vncoelho** [comment]
It is working, @shargon
Neo-cli is running here but the only problem is the need of LevelDb and its dependency `sudo apt-get update && sudo apt-get -y install libleveldb-dev`

**Comment by shargon** [comment]
> It is working, @shargon Neo-cli is running here but the only problem is the need of LevelDb and its dependency `sudo apt-get update && sudo apt-get -y install libleveldb-dev`

You can configure it with memory store by default.

**Comment by cschuchardt88** [comment]
@vncoelho 
`MemoryStore`

**Comment by cschuchardt88** [comment]
Looks like this a bug with it, looking to it

**Comment by cschuchardt88** [comment]
@vncoelho Guess I wrong about `null`
```json
    "Storage": {
      "Engine": "MemoryStore",
      "Path": null,
    },
```

**Comment by cschuchardt88** [comment]
Nope!! Its a bug with `neo-cli`

**Pull request merged**